modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -65,6 +65,8 @@ final case class Inputs(
 
   def add(extraElements: Seq[Element]): Inputs =
     if elements.isEmpty then this else copy(elements = (elements ++ extraElements).distinct)
+  def filter(filterElements: Seq[Element] => Seq[Element]): Inputs =
```

Comment:
This is more of a 'map' function judging by its signature, maybe `mapElements`? Or even something like `withElementsTransformed`, sicne we're not really mapping over the elements.